### ISSUE TITLE
Cypress npm install with snap Node.js on Ubuntu fails to install Cypress binary

### ISSUE THREAD

**Issue by MikeMcC399** [cypress-io/cypress-documentation#5868]
### Subject

Guides https://docs.cypress.io/guides/getting-started/installing-cypress#npm-install

### Description

If an attempt is made to install Cypress on Ubuntu using npm that has been installed via snap Node.js, then it fails to install the Cypress binary.

For example:

```shell
sudo snap install node --classic --channel=20
npm install cypress --save-dev
```

Typical log lines include output similar to the following:

```text
npm info run cypress@13.13.0 postinstall node_modules/cypress node index.js --exec install
npm info run cypress@13.13.0 postinstall { code: 1, signal: null }
...
npm error command sh -c node index.js --exec install
```

- This has been reported to https://github.com/nodejs/snap/issues/68 and the issue is awaiting a response.

If this is an on-going restriction, then it may need to be documented here. A workaround is to use:

```shell
npm install cypress --save-dev --foreground-scripts
```

Yarn is not affected. The following works correctly:

```shell
yarn add cypress -D
```

**Comment by jennifer-shehane** [comment]
Sounds fine to document for now.

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

> Sounds fine to document for now.

It will be a while until I get around to this. I hope there will be some feedback soon from the Node.js snap folks to understand the root cause and who could fix it. For the moment I just wanted to document the issue so there is a place to add any new facts.

**Comment by MikeMcC399** [comment]
- https://github.com/cypress-io/cypress-documentation/pull/5869 documents how to install Node.js. Ubuntu snap is not one of the methods recommended on the https://nodejs.org/en/download/ site, where Linux is listed generically. The next step would be to decide whether Ubuntu snap is given an explicit mention.

**Comment by MikeMcC399** [comment]
## Workaround Verification

Ubuntu `22.04.4` LTS

```shell
snap install node --classic --channel=20
echo node version $(node -v)
echo npm version $(npm -v)
rm -rf ~/.cache/Cypress
mkdir cy-snap-test
cd cy-snap-test
npm install cypress --save-dev --foreground-scripts
npx cypress verify
```

### Log

```text
$ snap install node --classic --channel=20
echo node version $(node -v)
echo npm version $(npm -v)
rm -rf ~/.cache/Cypress
mkdir cy-snap-test
cd cy-snap-test
npm install cypress --save-dev --foreground-scripts
npx cypress verify
node (20/stable) 20.15.1 from OpenJS Foundation (iojs✓) installed
node version v20.15.1
npm version 10.7.0

> cypress@13.13.0 postinstall
> node index.js --exec install

Installing Cypress (version: 13.13.0)

✔  Downloaded Cypress
✔  Unzipped Cypress
✔  Finished Installation /home/mike/.cache/Cypress/13.13.0

You can now open Cypress by running: node_modules/.bin/cypress open

https://on.cypress.io/installing-cypress


added 173 packages in 37s

39 packages are looking for funding
  run `npm fund` for details

✔  Verified Cypress! /home/mike/.cache/Cypress/13.13.0/Cypress
```

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

> Sounds fine to document for now.

- The feedback I got from issue https://github.com/nodejs/snap/issues/68 was essentially:
  > snaps have trouble doing child process execs because they are already executed in constrained environments

There wasn't any indication that the snap folks would look at resolving the Cypress installation issue in this environment, so there is no timeline at all for a resolution. It therefore makes sense to document the workaround. 

- The workaround is submitted as https://github.com/cypress-io/cypress-documentation/pull/5869 and I've bundled two section changes into one PR as they are so interrelated.
